test/extensions/filters/http/set_metadata/set_metadata_filter_test.cc

```diff
@@ -0,0 +1,152 @@
+#include <memory>
+
+#include "common/protobuf/protobuf.h"
+
+#include "extensions/filters/http/set_metadata/set_metadata_filter.h"
+#include "extensions/filters/http/well_known_names.h"
+
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/stream_info/mocks.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::_;
+using testing::ElementsAre;
+using testing::NiceMock;
+using testing::ReturnRef;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace SetMetadataFilter {
+
+class SetMetadataIntegrationTest : public testing::Test {
+
+public:
+  SetMetadataIntegrationTest() {}
+
+  void runFilter(envoy::config::core::v3::Metadata& metadata, const std::string& yaml_config,
+                 Http::TestRequestHeaderMapImpl& headers) {
+    envoy::extensions::filters::http::set_metadata::v3::Config ext_config;
+    TestUtility::loadFromYaml(yaml_config, ext_config);
+    auto config = std::make_shared<Config>(ext_config);
+    auto filter = std::make_shared<SetMetadataFilter>(config);
+
+    NiceMock<Http::MockStreamDecoderFilterCallbacks> decoder_callbacks;
+    NiceMock<Envoy::StreamInfo::MockStreamInfo> req_info;
+    filter->setDecoderFilterCallbacks(decoder_callbacks);
+
+    EXPECT_CALL(decoder_callbacks, streamInfo()).WillRepeatedly(ReturnRef(req_info));
+    EXPECT_CALL(req_info, dynamicMetadata()).WillRepeatedly(ReturnRef(metadata));
+    EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter->decodeHeaders(headers, true));
+    filter->onDestroy();
+  }
+
+  void check_key_int(ProtobufWkt::Struct const& s, const char* key, int val) {
```

Comment:
`checkKeyInt`

Reply:
Fixed in https://github.com/envoyproxy/envoy/pull/16400/commits/cfad0846aa208507d3e5450807bda6512209028c